pkg/service/store/store.go

```diff
@@ -133,3 +144,59 @@ func NewClusterService(name, namespace string) ClusterService {
 		Selector:  map[string]string{},
 	}
 }
+
+type clusterServiceObserver struct {
+	// merger is the interface responsible to merge service and
+	// endpoints into an existing cache
+	merger ServiceMerger
+
+	// swg provides a mechanism to know when the services were synchronized
+	// with the datapath.
+	swg *lock.StoppableWaitGroup
+}
+
+// OnUpdate is called when a service in a remote cluster is updated
+func (c *clusterServiceObserver) OnUpdate(key store.Key) {
+	if svc, ok := key.(*ClusterService); ok {
+		scopedLog := log.WithField(logfields.ServiceName, svc.String())
+		scopedLog.Debugf("Update event of cluster service %#v", svc)
+
+		c.merger.MergeClusterServiceUpdate(svc, c.swg)
+	} else {
+		log.Warningf("Received unexpected cluster service update object %+v", key)
+	}
+}
+
+// OnDelete is called when a service in a remote cluster is deleted
+func (c *clusterServiceObserver) OnDelete(key store.NamedKey) {
+	if svc, ok := key.(*ClusterService); ok {
+		scopedLog := log.WithField(logfields.ServiceName, svc.String())
+		scopedLog.Debugf("Delete event of cluster service %#v", svc)
+
+		c.merger.MergeClusterServiceDelete(svc, c.swg)
+	} else {
+		log.Warningf("Received unexpected cluster service delete object %+v", key)
+	}
+}
+
+// Controller for syncing services from the kvstore
+func JoinClusterServices(merger ServiceMerger) {
```

Comment:
Nit: godoc doesn't match up with the function name. Also could you expand a bit more on this one? The current comment doesn't give much info to me.

Reply:
Will do.